src/k8s-engine/wsl.ts

```diff
@@ -1143,73 +1158,100 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
               WSL_HELPER_BINARY: await this.getWSLHelperPath(),
               LOG_DIR:           logPath,
             });
-            await this.writeFile('/etc/logrotate.d/k3s', rotateConf, 0o644);
             await this.writeFile(`/etc/init.d/buildkitd`, SERVICE_BUILDKITD_INIT, 0o755);
             await this.writeFile(`/etc/conf.d/buildkitd`, SERVICE_BUILDKITD_CONF, 0o644);
+
             await this.runInit();
+            if (this.#currentContainerEngine === ContainerEngine.MOBY) {
+              await this.startService('docker', undefined);
+            } else {
+              await this.startService('containerd', undefined);
+            }
           }),
           this.progressTracker.action('Installing image scanner', 100, this.installTrivy()),
           this.progressTracker.action('Installing CA certificates', 100, this.installCACerts()),
           this.progressTracker.action('Installing helpers', 50, this.installWSLHelpers()),
-          this.progressTracker.action('Installing k3s', 100, async() => {
-            await this.deleteIncompatibleData(desiredVersion);
-            await this.installK3s(desiredVersion);
-            await this.persistVersion(desiredVersion);
-          }),
-        ]);
+        ];
+
+        if (enabledK3s) {
+          const actualDesiredVersion = desiredVersion as semver.SemVer;
+
+          installerActions.push(
+            this.progressTracker.action('Installing k3s', 100, async() => {
+              await this.deleteIncompatibleData(actualDesiredVersion);
+              await this.installK3s(actualDesiredVersion);
+              await this.persistVersion(actualDesiredVersion);
+            })
+          );
+        }
+        await Promise.all(installerActions);
 
         this.lastCommandComment = 'Running provisioning scripts';
         await this.progressTracker.action(this.lastCommandComment, 100, this.runProvisioningScripts());
 
-        await this.progressTracker.action('Starting k3s', 100,
-          this.startService('k3s', {
-            PORT:                   this.#desiredPort.toString(),
-            LOG_DIR:                await this.wslify(paths.logs),
-            'export IPTABLES_MODE': 'legacy',
-            ENGINE:                 this.#currentContainerEngine,
-            ADDITIONAL_ARGS:        this.cfg?.options.traefik ? '' : '--disable traefik',
-          }));
+        const k3sConf = {
+          PORT:                   this.#desiredPort.toString(),
+          LOG_DIR:                await this.wslify(paths.logs),
+          'export IPTABLES_MODE': 'legacy',
+          ENGINE:                 this.#currentContainerEngine,
+          ADDITIONAL_ARGS:        this.cfg?.options.traefik ? '' : '--disable traefik',
+        };
+
+        if (enabledK3s) {
+          await this.progressTracker.action('Starting k3s', 100,
+            this.startService('k3s', k3sConf));
+        } else {
+          await this.writeConf('k3s', k3sConf);
+        }
```

Comment:
Alternatively,
```suggestion
        await this.writeConf('k3s', k3sConf);
        if (enabledK3s) {
          await this.progressTracker.action('Starting k3s', 100,
            this.startService('k3s'));
        }
```
But it's fine as-is.

Reply:
I prefer the current form.